core/src/main/java/io/fabric8/maven/core/config/Profile.java

```diff
@@ -56,4 +102,30 @@ public ProcessorConfig getGeneratorConfig() {
         return generatorConfig;
     }
 
+    public int getOrder() {
+        return order;
+    }
+
+    @Override
+    public int compareTo(Profile o) {
+        int orderDiff = order - o.order;
+        if (orderDiff != 0) {
+            return orderDiff;
+        }
+        // Equals is only the very same object. This is ok, since at the end
+        // profiles are merge to a single profile
+        if (this.equals(o)) {
+            return 0;
+        } else {
+            // Try to determine the order based on creation time. The rule is, a newer profile
+            // is supposed to have a higher order.
+            long timeDiff = creationTime - o.creationTime;
```

Comment:
`creationTime` could quite often be equal here I assume so not sure it adds any value.

Can we work out an order from the various sources of profile?